src/dotnet/CarbonAware.Tools.WattTimeClient/IWattTimeClient.cs

```diff
@@ -0,0 +1,77 @@
+﻿using CarbonAware.Tools.WattTimeClient.Model;
+
+namespace CarbonAware.Tools.WattTimeClient;
+
+/// <summary>
+/// An interface for interacting with the WattTime API.
+/// </summary>
+public interface IWattTimeClient
```

Comment:
hi @bderusha and robert, I just wanted to gently re-voice here that 1 question/concern I had mentioned in the review yesterday...

my $0.02 (so far) is that the 'scenario' of us depending on a 3rd-party service (wattime) with an _unknown-change-rate OpenAPI spec_ seems to mostly match the "how-to" we actually published in [the nswag repo last year](https://github.com/RicoSuter/NSwag/blob/master/docs/tutorials/GenerateProxyClientWithCLI/generate-proxy-client.md) -- except for whether watttime org itself owns+maintains an openapi spec doc

Reply:
@dnastrain We chatted about this out-of-band but I wanted to capture the outcomes here:

- There is no officially maintained spec beyond the documentation page. (the link above is an old spec specifically for an alexa skill app)
- Even with a formal spec there are some inconsistencies about how the API actually works that would prevent us from being able to just use a generated client
- Having a custom client allows us to create a better experience for API users by improving the auth flow, overloading function calls to support more use-cases, and easier handling of some of the aforementioned inconsistencies.
- The v2 api has been stable for 4+ years with only backward compatible enhancements being added in that time

Reply:
agreed! -- and thx for recapping our offline chat!